django_filters/utils.py

```diff
@@ -78,8 +78,19 @@ def get_field_parts(model, field_name):
         >>> [p.verbose_name for p in parts]
         ['author', 'first name']
 
+        >>> parts = get_field_parts(Book, [author__first_name, 'surname'])
+        >>> [p.verbose_name for p in parts]
+        ['author', 'first_name', 'surname']
```

Comment:
`get_field_parts` is not intended to return a list of field names, but a list of the *individual parts* of a related field name (FK, m2m, 1to1). This result would be nonsensical, as `first_name` is a regular field on the author model, and does not represent a relationship to an additional `Name` model or something. 

Either way, as explained above, overloading `field_name` is not the correct path, and these changes are a symptom of that. I would revert these changes.

Reply:
You truth, I have read method descption about "the traversable relationships" and..ignored it.